### PR TITLE
Sea ice in RUC LSM 

### PR DESCRIPTION
**Pull request by tanyasmirnova**
## Description
Option for sea ice in RUC LSM is reinstated. Now sfc_drv_ruc.F90 has two calls: for land and sea ice.

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues should always be created before starting work on a PR branch!)
- fixes #<issue_number>
- fixes noaa-emc/fv3atm/issues/<issue_number>

## Testing

How were these changes tested?  
What compilers / HPCs was it tested with?  
Are the changes covered by regression tests? (If not, why? Do new tests need to be added?)  
Have the ufs-weather-model regression test been run? On what platform?  
- Will the code updates change regression test baseline? If yes, why? Please show the baseline directory below.
- Please commit the regression test log files in your ufs-weather-model branch

## Dependencies

If testing this branch requires non-default branches in other repositories, list them.
Those branches should have matching names (ideally)

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on noaa-emc/nems/pull/<pr_number>
- waiting on noaa-emc/fv3atm/pull/<pr_number>

### CODE REVIEW

File: gfsphysics/GFS_layer/GFS_typedefs.F90 [hunk 1]
```diff
@@ -1705,11 +1722,18 @@ real (kind=kind_phys), pointer :: prsik (:,:) => null()   !< Exner function at i
     real (kind=kind_phys), pointer      :: clouds(:,:,:)      => null()  !<
     real (kind=kind_phys), pointer      :: clw(:,:,:)         => null()  !<
     real (kind=kind_phys), pointer      :: clw_surf(:)        => null()  !<
+    real (kind=kind_phys), pointer      :: clw_surf_land(:) => null()  !<
```
Comment (DomHeinzeller) [inline gfsphysics/GFS_layer/GFS_typedefs.F90:1725]: There seem to be many variables that are added to both the Sfcprop and Interstitial DDT. Is this necessary? The interstitial variables aren't even allocated as far as I can see.

Comment (DomHeinzeller) [reply gfsphysics/GFS_layer/GFS_typedefs.F90:1725]: Or, asking the other way round, could we use the Interstitial DDT as much as possible to reduce the memory footprint?

Comment (tanyasmirnova) [reply gfsphysics/GFS_layer/GFS_typedefs.F90:1725]: Yes, we could. All RUC-specific variables are doubled now: for land and ice. I agree that we should do it the most efficient way.

Comment (DomHeinzeller) [reply gfsphysics/GFS_layer/GFS_typedefs.F90:1725]: Let's try to use the interstitial type then. We should make this switch a controlled experiment. After finalizing the ccpp-physics PR, run all the tests to create a new baseline and make sure we still get b4b reproducible results for warmstart runs. Then switch the variables only, nothing else, and verify against the baseline from before.

**Comment by DomHeinzeller** [comment]
This PR is superseded by https://github.com/NOAA-GSL/fv3atm/pull/56, do not merge.

**Comment by DomHeinzeller** [comment]
Replaced by https://github.com/NOAA-GSL/fv3atm/pull/60

**Pull request closed without merging**